### PR TITLE
[TR] [WIP] PIM-4246: Uniformize key of normalizers & setters

### PR DESCRIPTION
**Pull request by skeleton**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | Y
| CI currently passes? | ?
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y (todo)
| Fixed tickets        | PIM-4246
| DB schema updated?   | N
| Migration script?    | Y (todo)
| Doc PR               | N

### CODE REVIEW

File: upgrades/common/migrate_product_template.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        }
+    }
+
+    /**
+     * @return array
+     */
+    protected function getProductTemplates()
+    {
+        return $this->container->get('doctrine.orm.default_entity_manager')
+            ->getRepository('\Pim\Bundle\CatalogBundle\Entity\ProductTemplate')
```
Comment (jjanvier) [inline upgrades/common/migrate_product_template.php]: use the container param to retrieve the class

File: upgrades/common/migrate_product_template.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                            'scope'  => $value['scope'],
+                            'locale' => $value['locale'],
+                            'data'   => $value['value']
+                        ];
+                    }
+                }
+            }
+
+            if (!empty($data)) {
+                $sql = sprintf("UPDATE %s SET valuesData=:data WHERE id = :id", 'pim_catalog_product_template');
```
Comment (jjanvier) [inline upgrades/common/migrate_product_template.php]: put the name of the table at least in a const

File: UPGRADE-1.4.md [hunk 3]
```diff
@@ -19,6 +19,13 @@ The goal of this API is to give a straightforward and normalized way to update o
 
 To achieve a consistent API and avoid BC Breaks, we depreciate few methods from ProductUpdater.
 
+To have a better consistence between updaters and normalizers, `Pim\Bundle\TransformBundle\Normalizer\Structured\ProductValueNormalizer` returns now an array with a "data" key instead of "value" key.
+To convert database structure of 'pim_catalog_product_template', you can execute the following command in your project folder:
```
Comment (jjanvier) [inline UPGRADE-1.4.md]: This has an impact on the table `pim_catalog_product_template` which is used by the variant groups for instance. To convert the database structure of this table, you can execute the following command in your project folder:

**Comment by jjanvier** [comment]
GTM for me once fixed

File: UPGRADE-1.4.md [hunk 4]
```diff
@@ -19,6 +19,13 @@ The goal of this API is to give a straightforward and normalized way to update o
 
 To achieve a consistent API and avoid BC Breaks, we depreciate few methods from ProductUpdater.
 
+To have a better consistence between updaters and normalizers, `Pim\Bundle\TransformBundle\Normalizer\Structured\ProductValueNormalizer` returns now an array with a "data" key instead of "value" key.
```
Comment (grena) [inline UPGRADE-1.4.md]: returns now => now returns

**Pull request merged**